docs/launcher.md

```diff
@@ -1,106 +1,651 @@
+
+Ex:
+
+```yaml
 {
-  "status": <"started" | "stopped">,
-  "pid": 46910
+  "options": "--model TinyLlama/TinyLlama-1.1B-Chat-v1.0 --port 8005",
+  "env_vars": {
+    "VLLM_USE_V1": "1",
+    "VLLM_LOGGING_LEVEL": "DEBUG"
+  }
 }
+```
+
+### `VllmInstance`
+
+Represents a single vLLM instance with its process and configuration.
+
+**Key Methods:**
+
+- `start()`: Start the vLLM process
+- `stop(timeout=10)`: Stop the vLLM process gracefully (or force kill after timeout)
+- `is_running()`: Check if the process is currently running
+- `get_status()`: Get detailed status information
+
+#### `VllmMultiProcessManager`
+
+Manages multiple VllmInstance objects.
+
+**Key Methods:**
+
+- `create_instance(vllm_config, instance_id=None)`: Create and start a new instance
+- `stop_instance(instance_id, timeout=10)`: Stop a specific instance
+- `stop_all_instances(timeout=10)`: Stop all running instances
+- `get_instance_status(instance_id)`: Get status of a specific instance
+- `get_all_instances_status()`: Get status of all instances
+- `list_instances()`: List all instance IDs
+
+## Best Practices
+
+### 1. Port Management
+
+Each vLLM instance needs a unique port. Plan your port allocation:
 
+```bash
+# Good: Different ports
+Instance 1: --port 8000
+Instance 2: --port 8001
+Instance 3: --port 8002
+
+# Bad: Same port (will fail)
+Instance 1: --port 8000
+Instance 2: --port 8000  # ❌ Port conflict!
+```
+
+### 2. Instance Naming
+
+Use descriptive custom IDs for easier management:
+
+```bash
+# Good
+PUT /v2/vllm/instances/llama2-7b-chat
+PUT /v2/vllm/instances/gpt2-small-production
+PUT /v2/vllm/instances/opt-testing
+
+# Less clear
+POST /v2/vllm/instances  # Auto-generates UUID
 ```
+
+### 3. Graceful Shutdown
+
+Always delete instances when done to free resources:
+
+```bash
+# Delete specific instance
+curl -X DELETE http://localhost:8001/v2/vllm/instances/instance-id
+
+# Or clean up all instances
+curl -X DELETE http://localhost:8001/v2/vllm/instances
+```
+
+### 4. Error Handling
+
+Always check response status codes:
+
+```python
+response = requests.put(url, json=config)
+if response.status_code == 201:
+    print("Success:", response.json())
+elif response.status_code == 409:
+    print("Instance already exists")
+elif response.status_code == 500:
+    print("Failed to create vLLM instance:", response.json()["detail"])
+```
+
+### 5. Resource Limits
+
+Be mindful of system resources:
+
+- **Memory**: Each instance loads a full model into memory
+- **GPU**: Plan GPU allocation carefully
+- **CPU**: vLLM uses CPU for pre/post-processing
+- **Disk**: Models are cached in `~/.cache/huggingface`
+
+### 6. Testing
+
+Test with small models first:
+
+```bash
+# Use small models for testing
+--model facebook/opt-125m      # ~250MB
+--model gpt2                   # ~500MB
+
+# Then move to production models
+--model meta-llama/Llama-2-7b-hf   # ~14GB
+--model meta-llama/Llama-2-13b-hf  # ~26GB
+```
+
+---
+
+**Version:** 2.0
```

Comment:
Is this version number just a version of this document, or of something bigger? If the latter, what keeps it correctly updated here? If the former, when does it change?